src/redux/bancor2/apiData.slice.ts

```diff
@@ -0,0 +1,37 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { APIPool, APIToken, WelcomeData } from '../../services/api/bancor';
+import { Address } from '../../services/web3/types';
+
+interface ApiDataState {
+  isApiDataLoaded: boolean;
+  apiTokens: Map<Address, APIToken>;
+  apiPools: Map<Address, APIPool>;
+}
+
+export const initialState: ApiDataState = {
+  isApiDataLoaded: false,
+  apiTokens: new Map(),
+  apiPools: new Map(),
+};
+
+const apiDataSlice = createSlice({
+  name: 'apiData',
+  initialState,
+  reducers: {
+    setWelcomeData: (state, action: PayloadAction<WelcomeData>) => {
+      state.apiTokens = new Map(
```

Comment:
question: why not just set the raw data and use selectors to get the map?
we get this data every 15 seconds. We can easily tell that nothing changed and do nothing.
Only if it changed then next time the selectors will recalculate.
Currently we set the tokens and pools every 15 seconds

Reply:
yeah, I agree - see my comment above